guide/src/class.md

```diff
@@ -1166,16 +1171,16 @@ Python::with_gil(|py| {
 
         assert isinstance(square, cls)
         assert isinstance(square, cls.RegularPolygon)
-        assert square.side_count == 4
-        assert square.radius == 10.0
+        assert square._0 == 4
+        assert square._1 == 10.0
```

Comment:
Thank you very much for working on this! Just some initial thought, without having looked into the code yet.

I think I would find it more intuitive if tuple variants would use indexing on the Python side. 
```suggestion
        assert square[0] == 4
        assert square[1] == 10.0
```
This would be similar to how a normal tuple in Python would be accessed:
```python
square = (4, 10)
assert square[0] == 4
assert square[1] == 10
```
It might also solve the identifier problem you're having. What do you think about that? (`_0` does also use the "private" field convention, which is maybe not ideal, given that it is a normal public field)

Reply:
On second thought, we could also have that _additionally_ such that matching still works.

Reply:
I think generally we should advertise the tuple variants using indexing (here) and tuple match (below). Personally I think it looks nicer and in general I believe it is more intuitive and less confusing than referring to (kind of) arbitrary generated names.

We could definitely add a section (later) that mention these alternatives as well, but for the first introduction I think the above if more appropriate.

Reply:
Perfect 💯